lib/stack.go

```diff
@@ -0,0 +1,36 @@
+package lib
+
+// Stack type
+type Stack []string
+
+// IsEmpty check if stack is empty
+func (st *Stack) IsEmpty() bool {
+	return len(*st) == 0
+}
+
+// Push a new value onto the stack
+func (st *Stack) Push(str string) {
+	*st = append(*st, str) //Simply append the new value to the end of the stack
+}
+
+// Pop Remove top element of stack. Return false if stack is empty.
+func (st *Stack) Pop() bool {
+	if st.IsEmpty() {
+		return false
+	} else {
+		index := len(*st) - 1 // Get the index of top most element.
+		*st = (*st)[:index]   // Remove it from the stack by slicing it off.
+		return true
+	}
+}
+
+// Top Return top element of stack. Return false if stack is empty.
```

Comment:
```suggestion
// Top returns the top element of the stack. Returns empty string if stack is empty.
```

Reply:
Note that this isn't exactly the best interface because an empty string could be a real element.

Reply:
Yes you are right. I solved this issue in the latest commit.

Reply:
Yes you are right! I solved this by returning a boolean alongside the empty string if stack is empty
As far as empty string being a real element is concerned. The way we are deciding if the stack is empty is by using length. So it solves that issue.